tests/core/test_structured_data_signing.py

```diff
@@ -916,3 +916,170 @@ def test_unequal_array_dimension_between_schema_and_data():
         "`[{'wallet': '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB', 'name': 'Bob'}]` "
         "has dimensions `(1,)` whereas the schema has dimensions `(2, 'dynamic', 4)`"
     )
+
+
+def test_encode_structured_data_ignores_additional_data_in_a_custom_type():
+    message_with_additonal_data = {
+        "types": {
+            "EIP712Domain": [
+                {"name": "name", "type": "string"},
+            ],
+            "Person": [
+                {"name": "name", "type": "string"},
+            ],
+        },
+        "primaryType": "Person",
+        "domain": {
+            "name": "Name",
+        },
+        "message": {
+            "name": "Bob",
+            "pet": {
+                "animal": "cat",
+                "age": 3,
+            },
+        },
+    }
+
+    message_without_additonal_data = {
+        "types": {
+            "EIP712Domain": [
+                {"name": "name", "type": "string"},
+            ],
+            "Person": [
+                {"name": "name", "type": "string"},
+            ],
+        },
+        "primaryType": "Person",
+        "domain": {
+            "name": "Name",
+        },
+        "message": {
+            "name": "Bob",
+        },
+    }
+
+    assert encode_structured_data(
+        message_with_additonal_data
+    ) == encode_structured_data(message_without_additonal_data)
+
+
+def test_encode_typed_data_ignores_unused_types():
+    message_with_unused_type = {
+        "types": {
+            "EIP712Domain": [
+                {"name": "name", "type": "string"},
+            ],
+            "Person": [
+                {"name": "name", "type": "string"},
+            ],
+            "OtherPerson": [
+                {"name": "name", "type": "string"},
+            ],
+        },
+        "primaryType": "Person",
+        "domain": {
+            "name": "Name",
+        },
+        "message": {
+            "name": "Bob",
+        },
+    }
+
+    message_without_unused_type = {
```

Comment:
nit: (DRY) same as above... pop out the unused type

Reply:
dried